base_layer/core/src/transaction.rs

```diff
@@ -44,36 +58,124 @@ bitflags! {
     }
 }
 
+/// This macro implements the Ord, PartialOrd, PartialEq and Eq traits for the Hashable struct passed in
+macro_rules! hashable_ord {
```

Comment:
Perhaps this would be better implemented as a custom `[derive()]`, e.g
```
#[derive(HashEq, HashOrd)]
pub struct Transaction{
  ...
}

```

Reply:
Hmm, that does look better

Reply:
I made an issue for this to be done after this PR.

Reply:
Can't you do this with

```rust
impl<H: Hashable> Ord for H {
...
}
```

etc.?

Reply:
It feels like you should be able to but while attempting it I was getting some very strange error messages about conflicting implementation of the Trait on '&_' that I was not able to figure out. I will add this suggestion to the Issue about this.